projects/client/RabbitMQ.Client/src/client/api/IConnectionFactory.cs

```diff
@@ -136,6 +137,21 @@ public interface IConnectionFactory
         IConnection CreateConnection(IList<string> hostnames, String clientProvidedName);
 
         /// <summary>
+        /// Create a connection using a list of endpoints. By default each endpoint will be tried
+        /// in a random order until a successful connection is found or the list is exhausted.
+        /// The selection behaviour can be overriden by configuring the EndpointResolverFactory.
+        /// </summary>
+        /// <param name="endpoints">
+        /// List of endpoints to use for the initial
+        /// connection and recovery.
+        /// </param>
+        /// <returns>Open connection</returns>
+        /// <exception cref="BrokerUnreachableException">
+        /// When no hostname was reachable.
+        /// </exception>
```

Comment:
Should exceptions be documented on interfaces, or only on implementations? I've never been sure, and I've heard arguments both ways.

I guess you could also argue that some of the content in the `summary` tag is implementation specific.